kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java

```diff
@@ -2178,73 +2180,59 @@ public void ddl(ClassLoader loader, String resource)
 	 */
 	protected int prepareStatement(PreparedStatement pstmt, Object[] fields) throws SQLException, UnsupportedEncodingException
 	{
-		if (LOG.isDebugEnabled())
-		{
-			LOG.debug("prepareStatement(PreparedStatement " + pstmt + ", Object[] " + Arrays.toString(fields) + ")");
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("pstmt = {}, fields = {}", pstmt, Arrays.toString(fields));
 		}
 
 		// put in all the fields
 		int pos = 1;
-		if ((fields != null) && (fields.length > 0))
-		{
-			for (int i = 0; i < fields.length; i++)
-			{
-				if (fields[i] == null || (fields[i] instanceof String && ((String) fields[i]).length() == 0))
-				{
-					// treat a Java null as an SQL null,
-					// and ALSO treat a zero-length Java string as an SQL null
-					// This makes sure that Oracle vs MySQL use the same value
-					// for null.
+		if ((fields != null) && (fields.length > 0)) {
+			for (Object field : fields) {
+				if (field == null) {
+					// Treat a Java null as an SQL null.
+					// This makes sure that Oracle vs MySQL use the same value for null.
 					sqlServiceSql.setNull(pstmt, pos);
 
 					pos++;
```

Comment:
You could probably move this pos++ incrementer to the bottom of the for loop now too. It seems like they could have justed use i+1 in the original case.